src/seer/automation/codegen/models.py

```diff
@@ -194,7 +194,16 @@ class Comment(BaseModel):
         start_line: int
         suggestion: str | None = None
 
+    class PrDescription(BaseModel):
+        purpose: str
+        key_technical_changes: str
+        architecture_decisions: str
+        dependencies_and_interactions: str
+        risk_considerations: str
+        notable_implementation_details: str
```

Comment:
The `PrDescription` model has good structure but could benefit from field validation to ensure non-empty strings and reasonable length limits. Consider adding Pydantic field validators with `constr` type for string length constraints and non-empty validation.
```suggestion
class PrDescription(BaseModel):
    purpose: constr(min_length=1, max_length=2000)
    key_technical_changes: constr(min_length=1, max_length=2000)
    architecture_decisions: constr(min_length=1, max_length=2000)
    dependencies_and_interactions: constr(min_length=1, max_length=2000)
    risk_considerations: constr(min_length=1, max_length=2000)
    notable_implementation_details: constr(min_length=1, max_length=2000)
```